introduction_to_amazon_algorithms/jumpstart-foundation-models/text-generation-benchmarking/benchmarking/runner.py

```diff
@@ -0,0 +1,367 @@
+            **metrics_time,
+            "ProductionVariant": production_variant,
+            "PrimaryContainer": primary_container,
+        }
+
+    def run_single_predictor(
+        self,
+        model_id: str,
+        predictor: Predictor,
+        tokenizer_model_id: Optional[str] = None,
+    ) -> List[Dict[str, Any]]:
+        """Run benchmarker given a Predictor for an in-service model endpoint."""
+        metrics = []
+        try:
+            for payload_name, payload in self.payloads.items():
+                metrics_payload = self._run_benchmarking_tests(
+                    predictor, payload, model_id, payload_name, tokenizer_model_id
+                )
+                metrics.append(metrics_payload)
+        finally:
+            if self.clean_up is True:
+                self.clean_up_predictor(model_id, predictor)
+            else:
+                logging.info(f"{logging_prefix(model_id)} Skipping cleaning up resources ...")
+
+        return metrics
+
+    def run_single_model(self, model_id: str, model_args: Dict[str, Any]) -> Tuple[List[Dict[str, Any]], Predictor]:
+        """Run benchmarker for a single model.
+
+        If an `endpoint_name` is provided either as a key in `model_args` or saved in benchmarking metrics file from
+        a previous invocation of this benchmarker, then a predictor is attempted to be attached to this endpoint. If
+        an `endpoint_name` is not provided, then the model is deployed prior to benchmarking run.
+        """
+        endpoint_name = model_args.get("endpoint_name") or self.model_id_to_endpoint_name.get(model_id)
+        if endpoint_name is not None:
+            try:
+                predictor = self.retrieve_predictor_from_endpoint(endpoint_name, model_args)
+                logging.info(f"{logging_prefix(model_id)} Predictor successfully retrieved from endpoint name")
+            except Exception as e:
+                logging.warning(f"{logging_prefix(model_id)} Failed to retrieve predictor, re-deploying model: {e}")
+                predictor = self.deploy_model(model_id, model_args)
+        else:
+            pass
+            predictor = self.deploy_model(model_id, model_args)
+
+        self.model_id_to_endpoint_name[model_id] = predictor.endpoint_name
+        metrics = self.run_single_predictor(model_id, predictor, model_args["huggingface_model_id"])
+        return metrics, predictor
+
+    def retrieve_predictor_from_endpoint(
+        self, endpoint_name: str, model_args: Optional[Dict[str, Any]] = None
+    ) -> Predictor:
+        """Obtain a predictor from an already deployed endpoint."""
+        if model_args is not None:
+            jumpstart_model_args = model_args.get("jumpstart_model_args")
+            if jumpstart_model_args:
+                return retrieve_default(
+                    endpoint_name=endpoint_name,
+                    model_id=jumpstart_model_args["model_id"],
+                    model_version="*",
```

Comment:
consider reading from `jumpstart_model_args` as well and default to `*`

Reply:
changing to `jumpstart_model_args.get("model_version", "*")`